### PR TITLE
Support Picker field types

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Support different picker layouts by passing new fieldType prop. 
Current support `form` (default), `filter` and `settings` 

WOAUILIB-2267

## Changelog
Support Picker fieldType for different picker layouts. form (default), filter and settings

### CODE REVIEW

File: src/components/picker/types.tsx [hunk 1]
```diff
@@ -41,143 +47,145 @@ export interface PickerSearchStyle {
...
+    /**
+     * Temporary prop required for migration to Picker's new API
+     */
+    migrate?: boolean;
+    /**
+     * Temporary prop required for inner text field migration
+     */
+    migrateTextField?: boolean;
+    /* ...TextField.propTypes, */
+    fieldType?: PickerFieldTypes;
```
Comment (lidord-wix) [inline src/components/picker/types.tsx:61]: Missing description

Comment (lidord-wix) [inline src/components/picker/types.tsx:61]: Please update the api file with the new prop

File: demo/src/screens/componentScreens/PickerScreen.tsx [hunk 2]
```diff
@@ -289,6 +296,39 @@ export default class PickerScreen extends Component {
             style={{alignSelf: 'flex-start'}}
             onPress={() => this.picker.current?.openExpandable?.()}
           />
+
+          <Text text60 marginT-s5>
```
Comment (lidord-wix) [inline demo/src/screens/componentScreens/PickerScreen.tsx:300]: Add the `$textDefault` modifier for the dark mode support

File: demo/src/screens/componentScreens/PickerScreen.tsx [hunk 3]
```diff
@@ -289,6 +296,39 @@ export default class PickerScreen extends Component {
...
+            Different Field Types
+          </Text>
+          <Text text80 marginB-s5>(Form/Filter/Settings)</Text>
```
Comment (lidord-wix) [inline demo/src/screens/componentScreens/PickerScreen.tsx:303]: Add the `$textDefault` modifier for the dark mode support

File: src/components/picker/index.tsx [hunk 4]
```diff
@@ -196,6 +224,27 @@ const Picker = (props: PropsWithChildren<PickerProps> & ForwardRefInjectedProps
     useSafeArea
   ]);
 
+  const renderPickerInnerInput = () => {
+    if (fieldType === PickerFieldTypes.filter) {
+      return (
+        <Text text70 style={others.style}>
```
Comment (lidord-wix) [inline src/components/picker/index.tsx:230]: Add the `$textDefault` modifier for the dark mode support

File: src/components/picker/index.tsx [hunk 5]
```diff
@@ -196,6 +224,27 @@ const Picker = (props: PropsWithChildren<PickerProps> & ForwardRefInjectedProps
...
+          {label ?? others.placeholder}
+        </Text>
+      );
+    } else if (fieldType === PickerFieldTypes.settings) {
+      return (
+        <View row spread>
+          <Text text70 style={others.labelStyle}>
```
Comment (lidord-wix) [inline src/components/picker/index.tsx:237]: Add the `$textDefault` modifier for the dark mode support

File: src/components/picker/index.tsx [hunk 6]
```diff
@@ -142,6 +155,21 @@ const Picker = (props: PropsWithChildren<PickerProps> & ForwardRefInjectedProps
     ...pickerModalProps
   };
 
+  const propsByFieldType = useMemo(() => {
+    if (fieldType === PickerFieldTypes.filter) {
+      return {
+        preset: preset || null,
+        containerStyle: {flexDirection: 'row'},
+        trailingAccessory: <Icon source={dropdown}/>
```
Comment (lidord-wix) [inline src/components/picker/index.tsx:163]: The spacing between the icon and the text should be s1

File: src/components/picker/index.tsx [hunk 7]
```diff
@@ -222,15 +271,18 @@ const Picker = (props: PropsWithChildren<PickerProps> & ForwardRefInjectedProps
...
+            containerStyle={[paddings, margins, positionStyle, containerStyle, propsByFieldType?.containerStyle]}
             {...accessibilityInfo}
             importantForAccessibility={'no-hide-descendants'}
             value={label}
             selection={Constants.isAndroid ? {start: 0} : undefined}
             /* Note: Disable TextField expandable feature */
             topBarProps={undefined}
-          />
+          >
+            {renderPickerInnerInput()}
```
Comment (lidord-wix) [inline src/components/picker/index.tsx:284]: It feels a little bit an overkill to render the `filter` and `settings` types under a `TextField`, WDYT?

**Review by lidord-wix — COMMENTED** [review]
Looks good, I left a few comments :)

File: demo/src/screens/componentScreens/PickerScreen.tsx (hunk 3 above)
Comment (ethanshar) [reply demo/src/screens/componentScreens/PickerScreen.tsx:313]: Same answer as before

File: src/components/picker/index.tsx (hunk 4 above)
Comment (ethanshar) [reply src/components/picker/index.tsx:230]: same.. 
At this point, it just proves my first answer (:

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**